src/sqlfluff/dialects/dialect_bigquery.py

```diff
@@ -200,7 +200,10 @@
     SimpleArrayTypeGrammar=Sequence(
         "ARRAY",
         Bracketed(
-            Ref("DatatypeSegment"),
+            Sequence(
+                Ref("DatatypeSegment"),
+                Bracketed(Delimited(Ref("NumericLiteralSegment")), optional=True),
```

Comment:
Rather than repeating the code `Bracketed(Delimited(Ref("NumericLiteralSegment")), optional=True)` after each occurrence of `DatatypeSegment`, perhaps this should be added to the definition of `DatatypeSegment`. Because logically, `(10)` is part of the data type for `string(10)`.

Can you try this?

Reply:
Thanks for the good suggestion. Applied in [4eb54f5](https://github.com/sqlfluff/sqlfluff/pull/3735/commits/4eb54f5fa372d63322ad815917f2e681e5286c0a)